### PR TITLE
Cultists can now unanchor and move around cult structures

### PR DESCRIPTION
**Pull request by ChangelingRain**
:cl: Joan
rscadd: Cultists can now unanchor and reanchor cult structures by hitting them with a tome. Unanchored cult structures don't do anything.
/:cl:

If it's unanchored it won't do shit/can't be used, including pylons(see #17571 ; I didn't want to conflict with myself)
Trying to use it will give a 'you need a tome, bucko' message to cultists and do nothing for noncultists because why do _they_ care, anyway.

### CODE REVIEW

**Comment by Robustin** [comment]
Sure why not

File: code/game/gamemodes/cult/cult_structures.dm [hunk 1]
```diff
@@ -8,9 +8,21 @@
 
 /obj/structure/cult/examine(mob/user)
 	..()
+	user << "<span class='notice'>The [src] is [anchored ? "":"not "]secured to the floor.</span>"
```
Comment (phil235) [inline code/game/gamemodes/cult/cult_structures.dm]: \The [src] , or just [src] , but not The [src] or you'll end up with The the

File: code/game/gamemodes/cult/cult_structures.dm [hunk 2]
```diff
@@ -8,9 +8,21 @@
...
+/obj/structure/cult/attackby(obj/I, mob/user, params)
+	if(istype(I, /obj/item/weapon/tome) && iscultist(user))
+		anchored = !anchored
+		user << "<span class='notice'>You [anchored ? "":"un"]secure the [src] [anchored ? "to":"from"] the floor.</span>"
+		if(!anchored)
+			icon_state = "[initial(icon_state)]_off"
+		else
+			icon_state = initial(icon_state)
+		return
+	return ..()
```
Comment (phil235) [inline code/game/gamemodes/cult/cult_structures.dm]: else
	return ..()

this way you can remove the return on line 23

File: code/game/gamemodes/cult/cult_structures.dm [hunk 3]
```diff
@@ -8,9 +8,21 @@
...
 /obj/structure/cult/examine(mob/user)
 	..()
+	user << "<span class='notice'>The [src] is [anchored ? "":"not "]secured to the floor.</span>"
 	if(iscultist(user) && cooldowntime > world.time)
 		user << "<span class='cultitalic'>The magic in [src] is weak, it will be ready to use again in [getETA()].</span>"
 
+/obj/structure/cult/attackby(obj/I, mob/user, params)
+	if(istype(I, /obj/item/weapon/tome) && iscultist(user))
+		anchored = !anchored
+		user << "<span class='notice'>You [anchored ? "":"un"]secure the [src] [anchored ? "to":"from"] the floor.</span>"
```
Comment (phil235) [inline code/game/gamemodes/cult/cult_structures.dm]: same thing here: "the [src]"

File: code/game/gamemodes/cult/cult_structures.dm [hunk 4]
```diff
@@ -40,7 +55,7 @@
 			pickedtype = /obj/item/clothing/glasses/night/cultblind
 		if("Flask of Unholy Water")
 			pickedtype = /obj/item/weapon/reagent_containers/food/drinks/bottle/unholywater
-	if(pickedtype && Adjacent(user) && src && !qdeleted(src) && !user.incapacitated() && cooldowntime <= world.time)
+	if(anchored && pickedtype && Adjacent(user) && src && !qdeleted(src) && !user.incapacitated() && cooldowntime <= world.time)
```
Comment (phil235) [inline code/game/gamemodes/cult/cult_structures.dm]: also need to check if the user is still a cultist

File: code/game/gamemodes/cult/cult_structures.dm [hunk 5]
```diff
@@ -68,7 +86,7 @@
 			pickedtype = /obj/item/clothing/suit/hooded/cultrobes/berserker
 		if("Nar-Sien Hardsuit")
 			pickedtype = /obj/item/clothing/suit/space/hardsuit/cult
-	if(pickedtype && Adjacent(user) && src && !qdeleted(src) && !user.incapacitated() && cooldowntime <= world.time)
+	if(anchored && pickedtype && Adjacent(user) && src && !qdeleted(src) && !user.incapacitated() && cooldowntime <= world.time)
```
Comment (phil235) [inline code/game/gamemodes/cult/cult_structures.dm]: same, cultist check

File: code/game/gamemodes/cult/cult_structures.dm [hunk 6]
```diff
@@ -141,7 +162,7 @@
 			pickedtype = /obj/item/device/shuttle_curse
 		if("Veil Shifter")
 			pickedtype = /obj/item/device/cult_shift
-	if(pickedtype && Adjacent(user) && src && !qdeleted(src) && !user.incapacitated() && cooldowntime <= world.time)
+	if(anchored && pickedtype && Adjacent(user) && src && !qdeleted(src) && !user.incapacitated() && cooldowntime <= world.time)
```
Comment (phil235) [inline code/game/gamemodes/cult/cult_structures.dm]: culstist check

**Pull request merged**